renderer/src/helpers/utils/datetime-ui.ts

```diff
@@ -36,8 +36,10 @@ export function getWeekNumber(dateString: string) {
   const dateObj = getDateFromString(dateString);
   const startOfYear = new Date(dateObj.getFullYear(), 0, 1);
   const days = Math.floor((dateObj.getTime() - startOfYear.getTime()) / (24 * 60 * 60 * 1000));
+  const year = dateObj.getFullYear();
+  const leapYearDay = year === 2020 || year === 2024 || year === 2028 || year === 2032 ? 0 : 1;
```

Comment:
You harcoded years, right? could you create a function to calculate a leap year?